persistence/sql/persister_consent.go

```diff
@@ -135,59 +133,107 @@ func (p *Persister) GetForcedObfuscatedLoginSession(ctx context.Context, client,
 	return &s, nil
 }
 
+// CreateConsentRequest configures fields that are introduced or changed in the
+// consent request. It doesn't touch fields that would be copied from the login
+// request.
 func (p *Persister) CreateConsentRequest(ctx context.Context, req *consent.ConsentRequest) error {
-	return errorsx.WithStack(p.Connection(ctx).Create(req))
+	c, err := p.Connection(ctx).RawQuery(`
+UPDATE hydra_oauth2_flow
+SET
+	state = ?,
+	consent_challenge_id = ?,
+	consent_skip = ?,
+	consent_verifier = ?,
+	consent_csrf = ?
+WHERE login_challenge = ?;
+`,
+		flow.FlowStateConsentInitialized,
+		sqlxx.NullString(req.ID),
+		req.Skip,
+		req.Verifier,
+		req.CSRF,
+		req.LoginChallenge.String(),
+	).ExecWithCount()
+	if err != nil {
+		return sqlcon.HandleError(err)
+	}
+	if c != 1 {
+		return errorsx.WithStack(x.ErrNotFound)
+	}
+	return nil
 }
 
-func (p *Persister) GetConsentRequest(ctx context.Context, challenge string) (*consent.ConsentRequest, error) {
-	r := &consent.ConsentRequest{}
+func (p *Persister) GetFlowByConsentChallenge(ctx context.Context, challenge string) (*flow.Flow, error) {
+	f := &flow.Flow{}
+
+	if err := f.FindByConsentChallengeID(p.Connection(ctx), challenge); err != nil {
+		return nil, err
+	}
+
+	return f, nil
+}
 
-	if err := r.FindInDB(p.Connection(ctx), challenge); err != nil {
+func (p *Persister) GetConsentRequest(ctx context.Context, challenge string) (*consent.ConsentRequest, error) {
+	f, err := p.GetFlowByConsentChallenge(ctx, challenge)
+	if err != nil {
 		if errors.Is(err, sql.ErrNoRows) {
 			return nil, errorsx.WithStack(x.ErrNotFound)
 		}
 		return nil, sqlcon.HandleError(err)
 	}
 
-	return r, nil
+	return f.GetConsentRequest(), nil
 }
 
 func (p *Persister) CreateLoginRequest(ctx context.Context, req *consent.LoginRequest) error {
-	return errorsx.WithStack(p.Connection(ctx).Create(req))
+	f := flow.NewFlow(req)
+	return sqlcon.HandleError(p.Connection(ctx).Create(f))
+}
+
+func (p *Persister) GetFlow(ctx context.Context, challenge string) (*flow.Flow, error) {
+	var f flow.Flow
+	return &f, p.transaction(ctx, func(ctx context.Context, c *pop.Connection) error {
+		if err := (&f).FindInDB(c, challenge); err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				return errorsx.WithStack(x.ErrNotFound)
+			}
+			return sqlcon.HandleError(err)
+		}
+
+		return nil
+	})
 }
 
 func (p *Persister) GetLoginRequest(ctx context.Context, challenge string) (*consent.LoginRequest, error) {
-	var lr consent.LoginRequest
-	return &lr, p.transaction(ctx, func(ctx context.Context, c *pop.Connection) error {
-		if err := (&lr).FindInDB(c, challenge); err != nil {
+	var lr *consent.LoginRequest
+	return lr, p.transaction(ctx, func(ctx context.Context, c *pop.Connection) error {
+		f := &flow.Flow{}
+		if err := f.FindInDB(c, challenge); err != nil {
```

Comment:
To be honest, for readability, I think you can just do `c.Find(f, challenge)` here and delete `func (f *Flow) FindInDB(c *pop.Connection, id string) error`